src/routes/acquirerRoutes.ts

```diff
@@ -53,8 +54,20 @@ const skipThisStep = (step: string, formdata: FormData) => {
       const legalGatewayStep = formdata.steps["legal-gateway"].value as LegalGatewayStep
       return legalGatewayStep.yes.checked || legalGatewayStep.other.checked
     }
+    case "lawful-basis-personal": {
+      // Skip lawful-basis-personal if answer to data-type was "special" or "none"
+      const DataTypeStep = formdata.steps["data-type"].value as DataTypeStep
+      return DataTypeStep.special.checked || DataTypeStep.none.checked
+    }
+    case "lawful-basis-special": {
+      // Skip lawful-basis-special if answer to data-type was "personal" or "none"
+      const DataTypeStep = formdata.steps["data-type"].value as DataTypeStep
+      return DataTypeStep.personal.checked || DataTypeStep.none.checked
+    }
     case "role": {
-      return formdata.steps["data-type"].value === "none" || formdata.steps["data-type"].value === "";
+      // Skip role if answer to data-type was "none" or "null"
+      const DataTypeStep = formdata.steps["data-type"].value as DataTypeStep
+      return DataTypeStep.none.checked || formdata.steps["data-type"].value === "";
```

Comment:
formdata.steps["data-type"].value does not return "", what will return is special: {}, personal: {}, none: {}. I think you could write something like this: DataTypeStep.none.checked || (DataTypeStep.special.checked === undefined && DataTypeStep.personal.checked === undefined)